packages/shared_preferences/shared_preferences_foundation/lib/shared_preferences_foundation.dart

```diff
@@ -40,23 +41,38 @@ class SharedPreferencesFoundation extends SharedPreferencesStorePlatform {
 
   @override
   Future<bool> clear() async {
-    return clearWithPrefix(_defautPrefix);
+    return clearWithParameters(
+      ClearParameters(
+        filter: PreferencesFilter(prefix: _defaultPrefix),
+      ),
+    );
   }
 
   @override
-  Future<bool> clearWithPrefix(String prefix) async {
```

Comment:
This needs to stay, because older versions of the app-facing package can be used in combination with the new version of this package. We need to keep it, delegating to the new method.